src/Infra/Infra.Twitch/TwitchConnection.cs

```diff
@@ -30,19 +31,34 @@ public TwitchConnection(IDataProtection dataProtection)
             Client.OnMessageReceived += Client_OnMessageReceived;
             Client.OnWhisperReceived += Client_OnWhisperReceived;
             Client.OnConnected += Client_OnConnected;
+            Client.OnError += Client_OnError;
+            Client.OnConnectionError += Client_OnConnectionError;
+            Client.OnNoPermissionError += Client_OnNoPermissionError;
+        }
+
+        private void Client_OnNoPermissionError(object? sender, EventArgs e)
+        {
+        }
+
+        private void Client_OnConnectionError(object? sender, OnConnectionErrorArgs e)
+        {
+        }
+
+        private void Client_OnError(object? sender, OnErrorEventArgs e)
+        {
         }
 
         public void Connect(TwitchCredentials twitchCredentials)
         {
             ConnectionCredentials credentials = twitchCredentials.ToTwitchLib(_dataProtection);
-            Client.Initialize(credentials, twitchCredentials.Channel);
+            Client.Initialize(credentials, twitchCredentials.Channel.ToLowerInvariant());
 
             Client.Connect();
         }
 
         public void Disconnect()
         {
-            throw new NotImplementedException();
+            Client.Disconnect();
         }
 
```

Comment:
Does this imply that TwitchConnection needs to be `IDisposable` ?

Reply:
> Does this imply that TwitchConnection needs to be `IDisposable` ?

Very likely should be at some point. We also haven't fully set up the correct "closing" for the app. 🤣